flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/reader/splitreader/SplitReader.java

```diff
@@ -57,4 +57,11 @@
 	 * {@link #fetch()}.
 	 */
 	void wakeUp();
+
+	/**
+	 * Close the split reader.
+	 *
+	 * @throws Exception if closing the split reader failed.
+	 */
+	default void close() throws Exception {}
```

Comment:
I would personally not make this a default method, because it is a crucial one for every implementer to think about.

Reply:
I was about to ask for this :) otherwise, we may have resource/file leak for unclosed files in Iceberg source.